compiler/bindings/python/iree/build/executor.py

```diff
@@ -0,0 +1,517 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from typing import Callable, Collection, Generator, IO
+
+import abc
+import argparse
+import concurrent.futures
+import enum
+import inspect
+import multiprocessing
+import sys
+import time
+import traceback
+from pathlib import Path
+import threading
+
+_locals = threading.local()
+
+
+class FileNamespace(enum.StrEnum):
```

Comment:
`enum.StrEnum` was added in Python 3.11, so this code is failing to run on Python 3.10, which we still support

https://docs.python.org/3/library/enum.html
> Added in version 3.11: StrEnum, EnumCheck, ReprEnum, FlagBoundary, property, member, nonmember, global_enum, show_flag_values

Failing CI builds:
* Windows: https://github.com/iree-org/iree/actions/runs/11681252881/job/32525923299#step:8:1702
* macOS: https://github.com/iree-org/iree/actions/runs/11681217580/job/32525815632#step:9:1648

Reply:
Actually, we only publish wheels for Python 3.11 on Windows and macOS, so we could bump those workflows to use newer Python versions. I'll send a PR.